openwisp_controller/config/tests/test_admin.py

```diff
@@ -408,6 +410,22 @@ def test_clone_templates_operator_multi_orgs(self):
         response = self.client.post(path, post_data)
         self.assertContains(response, 'Clone templates')
 
+    def test_template_clonning_logentry(self):
+        t = self._create_template()
+        user = User.objects.create_superuser(
+            username='tester', password='tester', email='testeradmin@admin.com'
+        )
+        clone = t.clone(user)
+        ct = ContentType.objects.get(model='template')
+        LogEntry.objects.log_action(
+            user_id=user.id,
+            content_type_id=ct.pk,
+            object_id=clone.pk,
+            object_repr=clone.name,
+            action_flag=ADDITION,
+        )
+        self.assertIn('test-template (Clone)', str(LogEntry.objects.first()))
```

Comment:
@ManishShah120 it makes no sense to create the log entry in the test and then asserting it's created.
You have to simulate the usage of the clone action by sending an HTTP POST request, look at the other tests related to this feature in this file.

You can probably just edit `test_clone_template`.

Reply:
@nemesisdesign I have gone through other tests for performing the clone action with an HTTP POST request, but when we perform such operation in the tests scripts then no objects of `LogEntry` are created, so there is no way we can test the recent actions section in such manner. If we need the data in the `Recent Actions` sidebar then we will need to create the `LogEntry` manually after performing the `clone` operation and send a `get` request to assert that it gets created.

Reply:
Or if there is any other way we can achieve what you requested, then I am not aware of it. :thinking:

Reply:
@ManishShah120  instead of checking whether logs appear under `Recent Actions` section of Django admin site, you can check whether `LogEntry` objects are being created in the database. I did a quick test by trying to print all `LogEntry` objects created in `openwisp_controller.config.tests.test_admin.TestAdmin.test_clone_template` test and this was the output. 
```
<QuerySet [2021-02-20 06:25:22.021044+00:00, 2021-02-20 06:25:21.857934+00:00, 2021-02-20 06:25:21.758768+00:00]>
```

I don't understand the need of creating a Log Entry object in the test since the defies the whole point of testing.

Reply:
@pandafy I am getting different output:
![testadmin](https://user-images.githubusercontent.com/45564404/108587451-9a2fe800-7379-11eb-9a2c-858c1f04afe9.png)

Reply:
@pandafy the issue I was facing was due to a missing case in which I didn't implemented the logging option. So objects of LogEntry it weren't created during testing, Now I have added it, and for testing the `Recent Actions` I guess it will be better if we don't import `LogEntry` in the tests scripts. so I asserted with a get request. After all, this is where the users will get to see their recent actions.